### PR TITLE
DSD-1872: temp Header

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-1872](https://newyorkpubliclibrary.atlassian.net/browse/DSD-1872)

## This PR does the following:

- Adds the Header into the DS as a React component.
- Documentation is sparse because this is urgent, temporary, and only for DXP.

## How has this been tested?

- Locally in storybook and will test in Turbine through a release candidate.

## Accessibility concerns or updates

- N/A

### Accessibility Checklist

- [ ] Checked Storybook's "Accessibility" tab for color contrast and other issues.
- [ ] The feature works with keyboard inputs including tabbing back and forward and pressing space, enter, arrow, and esc keys.
- [ ] For hidden text or when `aria-live` is used, a screenreader was used to verify the text is read.
- [ ] For features that involve UI updates and focusing on DOM `ref`s, focus management was reviewed.
- [ ] The feature works when the page is zoomed in to 200% and 400%.

## Open Questions

-

### Checklist:

- [ ] I have updated the Storybook documentation and changelog accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

[DSD-1872]: https://newyorkpubliclibrary.atlassian.net/browse/DSD-1872?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/Header/Header.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import type { Meta, StoryObj } from "@storybook/react";
+
+import Header from "./Header";
+
+const meta: Meta<typeof Header> = {
+  title: "Components/Test/Header",
```
Comment (bigfishdesign13) [inline src/components/Header/Header.stories.tsx]: Nit: Maybe `"Components/Temp/Header"`?

File: src/components/Header/Header.mdx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+<Meta of={HeaderStories} />
+
+# Header
+
+## Overview
+
+Note: This is a temporary addition to Reservoir and should not be used in
+production unless told otherwise. Continue using the Header from
+`ds-header.nypl.org` and follow implementation instructions from the
+[nypl-header-app](https://github.com/NYPL/nypl-header-app) repository.
```
Comment (bigfishdesign13) [inline src/components/Header/Header.mdx]: You could also refer to the `Header` docs in Storybook.

https://nypl.github.io/nypl-design-system/reservoir/v3/?path=/docs/development-guide-header--docs

**Review by bigfishdesign13 — APPROVED** [review]
I did not review all of the files, but the component is functional.

**Comment by oliviawongnyc** [comment]
> I did not review all of the files, but the component is functional.

@EdwinGuzman same for me

**Review by bigfishdesign13 — APPROVED** [review]

**Pull request merged**